languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaRequestBuilderResourceRenderer.kt

```diff
@@ -95,10 +100,13 @@ class JavaRequestBuilderResourceRenderer @Inject constructor(val api: Api, overr
         this.pathArguments().forEach { requestArguments.add(it) }
 
         if(this.bodies != null && this.bodies.isNotEmpty()){
-            val methodBodyVrapType = this.bodies[0].type.toVrapType() as VrapObjectType
-            requestArguments.add(methodBodyVrapType.simpleClassName.decapitalize())
+            if(this.bodies[0].type.toVrapType() is VrapObjectType) {
+                val methodBodyVrapType = this.bodies[0].type.toVrapType() as VrapObjectType
+                requestArguments.add(methodBodyVrapType.simpleClassName.decapitalize())
+            }else {
+                requestArguments.add("jsonNode")
```

Comment:
Why do we need this here? The reason why we use `jsonNode` here is not very obvious and that why we add some docs here. Or extract the condition in the if to a local var with an meaningful name, may be like:
```kt
val isObjectType = this.bodies[0].type.toVrapType() is VrapObjectType
```
```